core/src/main/java/org/apache/iceberg/avro/PruneColumns.java

```diff
@@ -171,14 +241,19 @@ private static Schema copyRecord(Schema record, List<Schema.Field> newFields) {
     return copy;
   }
 
-  private static Schema.Field copyField(Schema.Field field, Schema newSchema) {
+  private static Schema.Field copyField(Schema.Field field, Schema newSchema, Integer fieldId) {
     Schema.Field copy = new Schema.Field(field.name(),
         newSchema, field.doc(), field.defaultVal(), field.order());
 
     for (Map.Entry<String, Object> prop : field.getObjectProps().entrySet()) {
       copy.addProp(prop.getKey(), prop.getValue());
     }
 
+    if (!AvroSchemaUtil.hasFieldId(field)) {
```

Comment:
Should this validate that the ID is the same if it is present?

Reply:
Updated